pkg/minikube/bootstrapper/images/images.go

```diff
@@ -64,14 +76,54 @@ func componentImage(name string, v semver.Version, mirror string) string {
 	return fmt.Sprintf("%s:v%s", path.Join(kubernetesRepo(mirror), name), v)
 }
 
+// fixes 13136 by getting the latest image version from the k8s.gcr.io repository instead of hardcoded
+func findLatestTagFromRepository(url string, lastKnownGood string) string {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", url, nil)
+
+	if err != nil {
+		klog.Infof("Failed to get latest image version for %s, reverting to version %s\n", url, lastKnownGood)
+		return lastKnownGood
+	}
+
+	resp, err := client.Do(req)
+
+	if err != nil || resp.StatusCode != http.StatusOK {
+		klog.Infof("Failed to get latest image version for %s, reverting to version %s\n", url, lastKnownGood)
+		return lastKnownGood
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		klog.Infof("Failed to read latest image version for %s, reverting to version %s\n", url, lastKnownGood)
+		return lastKnownGood
+	}
+
+	type TagsResponse struct {
+		Name string   `json:"name"`
+		Tags []string `json:"tags"`
+	}
+
+	tags := TagsResponse{}
+	err = json.Unmarshal(body, &tags)
+	if err != nil || len(tags.Tags) < 1 {
+		klog.Infof("Failed to read latest image version for %s, reverting to version %s\n", url, lastKnownGood)
```

Comment:
I feel if we error out here it's possible the JSON structure changed and could be valuable to include the `err` in the log, thoughts?

Could also argue the other errors could be important as well, but if we error on those likely the user just can't connect the internet and there's not much debugging could help with.

Reply:
fixed!

Reply:
> I feel if we error out here it's possible the JSON structure changed and could be valuable to include the `err` in the log, thoughts?
> 
> Could also argue the other errors could be important as well, but if we error on those likely the user just can't connect the internet and there's not much debugging could help with.

Yeah.. I think the err might be best included for all. Even if it's a network problem, or they need a proxy set, the err should Identity that for them. Will fix.